### PR TITLE
Dark mode color tweaks

### PR DESCRIPTION
**Pull request by VanAnderson**
Using functional colors for shadow on the overlay. I looked at what we were doing in dotcom, and it seems to simply be using `shadow.large`.

### Merge checklist
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/Overlay.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const widthMap = {
 /*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
 const StyledOverlay = styled.div<StyledOverlayProps & SystemCommonProps & SystemPositionProps & SxProp>`
   background-color: ${get('colors.bg.overlay')};
-  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
+  box-shadow: ${get('shadows.shadow.large')};
```
Comment (dgreif) [inline src/Overlay.tsx:37]: It looks like the old `Dialog` uses `large` (https://github.com/primer/components/blob/main/src/Dialog.tsx#L22), but the old `SelectMenu` uses `small` (https://github.com/primer/components/blob/main/src/SelectMenu/SelectMenuModal.tsx#L37).  Might be worth checking with @vdepizzol to see which would would prefer in the short term

**Comment by vdepizzol** [comment]
Hello @VanAnderson @dgreif! I'd like to make sure the Overlay patterns use the new shadows as proposed here: https://github.com/github/primer/issues/43.

I'd wait for functional variables to have updated values before merging this.

c/c @auareyou

**Comment by VanAnderson** [comment]
> Hello @VanAnderson @dgreif! I'd like to make sure the Overlay patterns use the new shadows as proposed here: [github/primer#43](https://github.com/github/primer/issues/43).
> 
> I'd wait for functional variables to have updated values before merging this.
> 
> c/c @auareyou

I think @pmarsceill mentioned this fix (https://github.com/primer/components/issues/1278) should be prioritized for the upcoming Memex release. Unless https://github.com/github/primer/issues/43 is right around the corner, do you think we could merge this and then follow up to update it with the new functional vars?

**Comment by vdepizzol** [comment]
> I think @pmarsceill mentioned this fix (#1278) should be prioritized an upcoming release of the project they are working on. Unless github/primer#43 is right around the corner, do you think we could merge this and then follow up to update it with the new functional vars?

👍  Absolutely! 

@auareyou do you have a timeline of when we can fix the shadows in primer primitives?

**Comment by auareyou** [comment]
I'm on it! We'll get this updated today

**Comment by vdepizzol** [comment]
@VanAnderson for reference, @simurai is working on a Primer Primitives here: https://github.com/primer/primitives/pull/135

**Comment by VanAnderson** [comment]
> I'm on it! We'll get this updated today

Awesome, thanks @vdepizzol! @auareyou if you could ping me in slack once you complete that work I will make sure to expedite getting this work in :)

**Review by colebemis — APPROVED** [review]

**Pull request merged**